store/driver/txn/batch_getter.go

```diff
@@ -16,11 +16,54 @@ package txn
 
 import (
 	"context"
+	"unsafe"
 
-	"github.com/pingcap/errors"
 	"github.com/pingcap/tidb/kv"
+	tikverr "github.com/tikv/client-go/v2/error"
+	"github.com/tikv/client-go/v2/txnkv/transaction"
 )
 
+// tikvBatchGetter is the BatchGetter struct for tikv
+type tikvBatchGetter struct {
+	tidbBatchGetter BatchGetter
+}
+
+func (b tikvBatchGetter) BatchGet(ctx context.Context, keys [][]byte) (map[string][]byte, error) {
+	// toTiDBKeys
+	kvKeys := *(*[]kv.Key)(unsafe.Pointer(&keys))
+	vals, err := b.tidbBatchGetter.BatchGet(ctx, kvKeys)
+	return vals, err
+}
+
+// tikvBatchBufferGetter is the BatchBufferGetter struct for tikv
```

Comment:
Could you please explain more clearly about why we need this struct and how this struct works?

Reply:
We need to call `NewBufferBatchGetter` method in client-go. The type of the parameters to this method is different from TiDB. For example, the BatchGetter in TiDB is an interface that implements `BatchGet(ctx context.Context, keys []kv.Key)`, but in client-go, the BatchGetter is an interface that implements `BatchGet(ctx context.Context, keys [][]byte)`. So we need to abstract out a new class to implement this function `BatchGet(ctx context.Context, keys [][]byte)`.